src/SimpleExecutionController.cpp

```diff
@@ -102,10 +108,19 @@ int SimpleExecutionController::main() {
     auto htcondor_compute_service = *this->htcondor_compute_services.begin();
 
 
+    // Shuffle jobs for submission
+    std::vector<std::string> job_spec_keys;
```

Comment:
One could also think of making this rather an ID (unsigned integer). This would probably save some runtime and memory, when one need not compare the strings for every job.

Reply:
Hm I don't see an urgent need in that since the keys are only used to access the job_specs in the corresponding map.

This means, they are anyhow handled efficiently for this (as far as I know, maps create internally hashes to access information & compare keys).

And in case of shuffling of a vector, a permutation is performed, based on indices.

So I don't see a performance gain here, at least concerning CPU time. For memory, I agree. Additional vector consumes space. Maybe it is possible to save the references to the keys.